bgs/bgs.go

```diff
@@ -106,7 +106,25 @@ type SocketConsumer struct {
 	EventsSent  promclient.Counter
 }
 
-func NewBGS(db *gorm.DB, ix *indexer.Indexer, repoman *repomgr.RepoManager, evtman *events.EventManager, didr did.Resolver, rf *indexer.RepoFetcher, hr api.HandleResolver, ssl bool, compactInterval time.Duration) (*BGS, error) {
+type BGSConfig struct {
+	SSL               bool
+	CompactInterval   time.Duration
+	DefaultRepoLimit  int64
+	ConcurrencyPerPDS int64
+	MaxQueuePerPDS    int64
+}
+
+func DefaultBGSConfig() BGSConfig {
+	return BGSConfig{
+		SSL:               true,
+		CompactInterval:   4 * time.Hour,
+		DefaultRepoLimit:  100,
+		ConcurrencyPerPDS: 100,
+		MaxQueuePerPDS:    1_000,
+	}
+}
+
+func NewBGS(db *gorm.DB, ix *indexer.Indexer, repoman *repomgr.RepoManager, evtman *events.EventManager, didr did.Resolver, rf *indexer.RepoFetcher, hr api.HandleResolver, config BGSConfig) (*BGS, error) {
```

Comment:
This should probably receive the config as a pointer and then init it as the default if the config it receives is nil.

Reply:
good point, updated to to that